docs/USING_PUBLIC_CLOUDS.rst

```diff
@@ -0,0 +1,374 @@
+
+Make sure versioning is enabled in your remote AWS S3 hosted bucket. To check,
+using the AWS Console, click on your bucket name, then on "Properties" at the
+top, and then you should see something like this:
+
+.. figure:: ../res/aws-console-versioning-enabled.png
+   :alt: AWS Console showing versioning enabled
+
+Microsoft Azure as a data backend
+---------------------------------
+
+From the MS Azure Console
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+From your Storage Account dashboard, create a container where you will host your
+data for this new location constraint.
+
+You will also need to get one of your Storage Account Access Keys, and to
+provide it to CloudServer.
+This can be found from your Storage Account dashboard, under "Settings, then
+"Access keys".
+
+In this example, our container will be named ``zenkontainer``, and will belong
+to the ``zenkomeetups`` Storage Account.
+
+From the CloudServer repository
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+locationConfig.json
+^^^^^^^^^^^^^^^^^^^
+
+Edit this file to add a new location constraint. This location constraint will
+contain the information for the MS Azure container to which you will be writing
+your data whenever you create a CloudServer bucket in this location.
+There are a few configurable options here:
+
+- :code:`type` : set to :code:`azure` to indicate this location constraint is
+  writing data to MS Azure;
+- :code:`legacyAwsBehavior` : set to :code:`true` to indicate this region should
+  behave like AWS S3 :code:`us-east-1` region, set to :code:`false` to indicate
+  this region should behave like any other AWS S3 region (in the case of MS Azure
+  hosted data, this is mostly relevant for the format of errors);
+- :code:`azureBlobEndpoint` : set to your storage account's endpoint, usually
+  :code:`https://{{storageAccountName}}.blob.core.windows.name`;
+- :code:`azureContainerName` : set to an *existing container* in your MS Azure
+  storage account; this is the container in which your data will be stored for
+  this location constraint;
+- :code:`bucketMatch` : set to :code:`true` if you want your object name to be
+  the same in your local bucket and your MS Azure container; set to
+  :code:`false` if you want your object name to be of the form
+  :code:`{{localBucketName}}/{{objectname}}` in your MS Azure container ;
+- :code:`azureStorageAccountName` : the MS Azure Storage Account to which your
+  container belongs;
+- :code:`azureStorageAccessKey` : one of the AccessKeys associated to the above
+  defined MS Azure Storage Account.
+
+.. code:: json
+
+    (...)
+    "azure-test": {
+	"type": "azure",
+        "legacyAwsBehavior": false,
+        "details": {
+          "azureBlobEndpoint": "https://zenkomeetups.blob.core.windows.net/",
+	  "bucketMatch": true,
+	  "azureStorageAccountName": "zenkomeetups",
+	  "azureStorageAccessKey": "auhyDo8izbuU4aZGdhxnWh0ODKFP3IWjsN1UfFaoqFbnYzPj9bxeCVAzTIcgzdgqomDKx6QS+8ov8PYCON0Nxw=="
+	}
+    },
+    (...)
```

Comment:
Optionally, users may set the azureStorageAccountName, azureStorageAccessKey, and azureStorageEndpoint as environment variables in the following form:
```
$ export azure-test_AZURE_STORAGE_ACCOUNT_NAME="zenkomeetups"
$ export azure-test_AZURE_STORAGE_ACCESS_KEY="auhyDo8izbuU4aZGdhxnWh0ODKFP3IWjsN1UfFaoqFbnYzPj9bxeCVAzTIcgzdgqomDKx6QS+8ov8PYCON0Nxw=="
$ export azure-test_AZURE_STORAGE_ENDPOINT="https://zenkomeetups.blob.core.windows.net/"
```
If they specify these details in both environmental variables and the locationConfig.json, the values stored in the environmental variables take priority and are the ones that will be used.

I don't know if it's important to document this detail, it could be we created these environmental variables primarily for our own testing purposes. But since we have this option...

Reply:
I added a "TIP" box, I think it's relevant indeed